provider/export/src/locale_family.rs

```diff
@@ -198,52 +198,50 @@ impl Writeable for LocaleFamilyAnnotations {
     }
 }
 
-/// An error while parsing a [`LocaleFamily`].
+/// An error while parsing a [`DataLocaleFamily`].
 #[derive(Debug, Copy, Clone, PartialEq, Display)]
 #[non_exhaustive]
 pub enum LocaleFamilyParseError {
-    /// An error bubbled up from parsing a [`LanguageIdentifier`].
+    /// An error bubbled up from parsing a [`DataLocale`].
     #[displaydoc("{0}")]
-    LanguageIdentifier(ParseError),
+    Locale(ParseError),
     /// Some other error specific to parsing the family, such as an invalid lead byte.
     #[displaydoc("Invalid locale family")]
     InvalidFamily,
 }
 
 impl From<ParseError> for LocaleFamilyParseError {
     fn from(err: ParseError) -> Self {
-        Self::LanguageIdentifier(err)
+        Self::Locale(err)
     }
 }
 
 impl std::error::Error for LocaleFamilyParseError {}
 
-impl FromStr for LocaleFamily {
+impl FromStr for DataLocaleFamily {
     type Err = LocaleFamilyParseError;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         if s == "full" {
             return Ok(Self::FULL);
         }
-        let (first, remainder) = s
-            .as_bytes()
-            .split_first()
-            .ok_or(LocaleFamilyParseError::InvalidFamily)?;
+        let mut iter = s.chars();
```

Comment:
Observation: Fine for another PR, but we should allow this to parse from bytes instead of str

Reply:
Yeah I'd like for `DataLocale` to get an API surface comparable to `LanguageIdentifier`.